components/reference-implementation-openbanking-nextgenpsd2/src/main/java/org/wso2/openbanking/nextgenpsd2/extensions/api/impl/ConsentManagementAPIImpl.java

```diff
@@ -0,0 +1,228 @@
+        String requestId = requestBody.getRequestId();
+        try {
+            ConsentManagementValidationHandler consentHandler = CommonConsentValidationUtil
+                    .getConsentManagementResponseHandler(requestBody.getData().getConsentResourcePath());
+
+            SuccessResponseForResponseAlternation validationResponse;
+
+            if (consentHandler != null) {
+                validationResponse = consentHandler.enrichCreationResponse(requestBody);
+            } else {
+                // Server error since if path is invalid consent creation should have failed
+                // thus making this unreachable
+                JSONObject errorResponse = ErrorUtil.getFormattedErrorResponse(ErrorUtil.constructBerlinError(null,
+                        TPPMessage.CategoryEnum.ERROR, null, ErrorConstants.PATH_INVALID));
+                return Response.status(Response.Status.BAD_REQUEST).entity(errorResponse).build();
+            }
+
+            return Response.ok().entity(new JSONObject(validationResponse).toString()).build();
+
+        } catch (BadRequestException | ServerErrorException e) {
+            log.error("[" + requestId + "] " + "An error occurred creating consent.", e);
+            return Response.status(e.getStatus()).entity(e.getFormattedErrorAsString()).build();
+        } catch (org.wso2.openbanking.nextgenpsd2.extensions.exceptions.ApiException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Method for returning the response for pre-processing consent creation request.
+     *
+     * @param requestBody
+     * @return
+     */
+    public static Response preProcessConsentCreation(PreProcessConsentCreationRequestBody requestBody) {
+        String requestId = requestBody.getRequestId();
+        try {
+            // Validate X-request-ID header
+            // Enable forwarding of the specific header in accelerator configurations
+            CommonConsentValidationUtil.validateRequestIdentificationHeader(requestBody.getData().getRequestHeaders());
+
+            // Extract consent initiation data
+            JSONObject consentInitiationDataJSON;
+            try {
+                consentInitiationDataJSON = CommonConsentValidationUtil.convertObjectToJson(requestBody.getData()
+                        .getConsentInitiationData());
+
+                if (consentInitiationDataJSON.isEmpty()) {
+                    // If payload is empty
+                    JSONObject errorResponse = ErrorUtil.getFormattedFailedResponse(400,
+                            ErrorUtil.constructBerlinError(null, TPPMessage.CategoryEnum.ERROR,
+                                    TPPMessage.CodeEnum.FORMAT_ERROR, ErrorConstants.PAYLOAD_NOT_PRESENT_ERROR));
+                    return Response.ok().entity(errorResponse.toString()).build();
+                }
+
+            } catch (JSONException e) {
+                // If payload is not JSON
```

Comment:
Doesn't this handles by the `@Consumes({ "application/json" })` annotation in the preProcessConsentCreationAPI class?